### ISSUE TITLE
[Feature request] We would like to be able to override the release version while generating manifest

### ISSUE THREAD

**Issue by Kaixiang** [cloudfoundry-incubator/routing-release#63]
Hi:

It will be great if we could have a separate stub file which we can specify versions for routing/cf releases, instead of using [latest version](https://github.com/cloudfoundry-incubator/routing-release/blob/develop/templates/routing.yml#L5) all the times.

This will make the automation process easier to dictate the version it wants to deploy.

Thoughts?

**Comment by shalako** [comment]
Hello @Kaixiang 
I'll prioritize the story for eng review.

**Comment by Kaixiang** [comment]
Thanks @shalako !

**Comment by shalako** [comment]
Questions from the team:
- Why would you deploy anything other than latest?
- You mention `routing/cf`; do you want this feature for `routing-release` or `cf-release` or both?
- Team is not aware of other releases supporting this feature; if you are aware of one could you provide an example?

Thank you!

**Comment by Kaixiang** [comment]
@shalako 

Below are few cents:

1. There will always be compatibility issue between releases giving a long enough time range. 
As for operation team, we do want to have the ability to control which version of the release goes to production. And this is also a capability we would like to have when we have to roll back to previous releases when something bad happens.

 2. Sorry for the misleading, I meant to configure the [cf version](https://github.com/cloudfoundry-incubator/routing-release/blob/develop/templates/routing.yml#L7) and [routing-release](https://github.com/cloudfoundry-incubator/routing-release/blob/develop/templates/routing.yml#L5) for routing deployment.

3. Diego release support this feature, here is an example for the [Manifest generation script](https://github.com/cloudfoundry/diego-release/blob/afbc2009f81d6348091f7681c1f9c0a21f971908/scripts/generate-deployment-manifest#L22)

**Comment by aaronshurley** [comment]
@shalako do you need anything further or can we prioritize this issue?

Thanks!

**Comment by abbyachau** [comment]
Prioritized for eng review again

**Comment by coreyti** [comment]
Hi @shalako,

In response to your earlier question, what we understand to be "best practice" is to be specific about which versions of BOSH releases are available and in use. The BOSH team recommends *against* using "latest". Note too that, when using a BOSH Director with multiple deployments (US-PWS Director manages almost 50), the releases are shared across deployments and often uploaded out-of-band of a particular `bosh deploy`. This leads to a greater likelihood of deploying an unexpected version somewhere when using "latest".

**Comment by shashwathi** [comment]
@coreyti @Kaixiang : 

Thank you for submitting this issue. This feature should be available in the next final release of routing(v0.158.0). To override the release version, stub should in the following format

```
---
release_version_overrides:
- name: routing
  version: 0.157.0
- name: cf
  version: 265
```

Closing this issue for now. Feel free to open another issue if run into other problems. 

Regards
Shash
